build.sh

```diff
@@ -3,10 +3,15 @@ set -e
 
 # To build with tests, install cmocka-dev and define EXTRA_CMAKE_FLAGS as:
 # EXTRA_CMAKE_FLAGS=-DUNIT_TESTING=ON
+# Also make sure that gcc-8 is used when building the test suite.  If it isn't, add (e.g.) -DCMAKE_C_COMPILER=gcc-8 -DCMAKE_CXX_COMPILER=g++-8
```

Comment:
When I first tried building the unit tests, I was getting errors when using certain -W flags that I thought were only supported on gcc-8+ (of course, now, I can't recall exactly which flags).  At that time, I was also having trouble with cmocka (the supplied binary version from Ubuntu 18.04 was too downlevel so I had to build cmocka from source).  But when I went to reproduce the error (I thought I had with gcc-7) everything builds just fine.  I may have also upgraded by gcc-7 since then (I currently have 7.4.0 installed).  Bottom line is I cannot reproduce the problem I thought I had with gcc-7.